Lesson6.ipynb

```diff
@@ -27,7 +34,686 @@
+    "p3_daily_steps = [16272, 17231, 20595, 17047, 15216, 42590, 10969, 20687, 19170, 12703, 17192, 12865, 10960, 9105, 16019, 12646, 10042, 13353, 16072, 41673, 13425, 11262, 7801, 6666, 5276, 11353, 5344, 6282]\n",
+    "p4_daily_steps = [10233, 16120, 12897, 24680, 13060, 20489, 10230, 25565, 10029, 12696, 13938, 9475, 5297, 8573, 9857, 15341, 9482, 11649, 5804, 11080, 6245, 7611, 8401, 5596, 6491, 7637, 7610, 9130]\n",
+    "p5_daily_steps = [14126, 14110, 10111, 20440, 21416, 16989, 25371, 21539, 23045, 20043, 20328, 12058, 12004, 3301, 9789, 6671, 7893, 9589, 10459, 5091, 6329, 6784, 6543, 17984, 13588, 11077, 7856, 20897]\n",
+    "\n",
+    "team = { # a dictionary mapping initial to list of daily steps\n",
+    "    'G': p1_daily_steps, \n",
+    "    'I': p2_daily_steps, \n",
+    "    'P': p3_daily_steps, \n",
+    "    'T': p4_daily_steps, \n",
+    "    'D': p5_daily_steps\n",
+    "}\n",
+    "\n",
+    "# Print minimum and maximum no. of steps walked by each person in this format: 'G's (minimum, maximum) steps per day are: (534, 24586)'\n",
+    "# Hint 1: convert list to numpy array and apply amin and amax functions. \n",
+    "# Hint 2: If a dictionary is called dict, dict.keys() will return the list of keys in that dictionary.\n",
+    "\n",
+    "# Start here\n",
+    "\n",
+    "\n",
+    "\n",
+    "\n",
+    "# Print the person with heighest average and his total no. of steps\n",
+    "#Hint 1: use numpy's average() and sum() functions\n",
+    "\n",
+    "# Start here\n",
+    "\n",
+    "\n",
+    "\n",
+    "\n",
+    "# Print the percentage increment between first two days and last two days of each team member\n",
+    "# Hint 1: (total_steps_in_first_2_days - total_steps_in_last_2_days)*100/total_steps\n",
+    "#Start here\n",
+    "\n",
+    "\n",
+    "\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## What Is Object-Oriented Programming?\n",
+    "\n",
+    "So far, We have learned basic fundamentals in Python language. Now, let's focus on how to build a house from what we have learned. \n",
+    "\n",
+    "Object-oriented programming (OOP) is a programming model based on the concept of \"objects\", which may contain data, in the form of attributes and their behavior in the form of methods. \n",
+    "\n",
+    "A feature of objects is that an object's methods can access and often modify the data fields of the object with which they are associated (objects have a notion of \"this\" or \"self\"). In OOP, programs are designed by making them out of objects that interact with one another.\n",
+    "\n",
+    "Each Object has its own attributes, and behavior. Objects are separate from one another. They have their own existence, their own identity that is independent of other objects.\n",
```

Comment:
"Objects are separate from one another. They have their own existence, their own identity that is independent of other objects."
Should be read
"Objects are separated from one another. They have their own existence, their own identity which is independent of other objects."